slack_bolt/listener_matcher/builtins.py

```diff
@@ -176,6 +177,17 @@ def _verify_message_event_type(event_type: str) -> None:
         raise ValueError(error_message_event_type(event_type))
 
 
+def function_event(
```

Comment:
Can we add a docstring here to clarify what this is for? It is clear to me given the context, but I don't trust my memory to remember this in e.g. a year from now 😅 

The event and construct here (Slack Custom Function) is also named in a way that makes it confusing when discussing it in a programming concept (are you talking about the Slack Custom Function, or a Python function? etc.). I think we need to be disciplined about terminology.

Reply:
Technically the `type` of the specific event related here is called a `function_executed` event - perhaps tweaking the naming to match that makes sense too.

Reply:
I like renaming this function to match the event `type` 💯  I agree that it aligns better with the terminology